plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -833,8 +833,8 @@ def modify_xml(self):
         self._templatize_pkgid(filelists)
 
         self.repodata['primary'] = utils.remove_fake_element(utils.element_to_text(faked_primary))
-        self.repodata['other'] = utils.remove_fake_element(utils.element_to_text(other))
-        self.repodata['filelists'] = utils.remove_fake_element(utils.element_to_text(filelists))
+        self.repodata['other'] = utils.element_to_text(other)
+        self.repodata['filelists'] = utils.element_to_text(filelists)
```

Comment:
Overzealous is probably a good descriptor. If we were using the ``faked_other`` and ``faked_filelists`` objects here, then we would need to run it through the ``remove_fake_element`` code.

The main reason the ``remove_fake_element`` function exists is because the ``primary`` snippets use an xml namespace that makes it difficult to render just part of an xml document. So we have to render the whole thing, let the namespace be declared in the fake element, then slice the fakeness away.

In theory we could make ``remove_fake_element`` somehow guard against the possibility that it will be used incorrectly again in the future, but I'm not sure if it's worth it. It could raise an exception if there's no fake element found, or just not do any slicing in that case.